### PR TITLE
Basic support for Paged Results Control (RFC 2696)

### PR DESCRIPTION
**Pull request by broeng**
This implements rudimentary support for the Paged
Results Control.

No actual pagination is performed, and we ignore
any requests for specific window sizes for paginated results.

Instead, the full list of search results is returned for any searches, and a control is added to the
SearchResultsDone message, informing the client that there is no further results available.

### CODE REVIEW

**Comment by broeng** [comment]
Haven't tested yet. Only pushed to gather feedback.

File: server/src/infra/ldap_handler.rs [hunk 1]
```diff
@@ -188,7 +188,8 @@ fn root_dse_response(base_dn: &str) -> LdapOp {
             },
             LdapPartialAttribute {
                 atype: "supportedControl".to_string(),
-                vals: vec![],
+                // Paged Results Control (RFC 2696)
+                vals: vec![b"1.2.840.113556.1.4.319".to_vec()],
```
Comment (nitnelave) [inline server/src/infra/ldap_handler.rs:192]: Hmmm... I wouldn't go as far as listing it in the supported controls. Our implementation is... rudimentary :D

File: server/src/infra/ldap_server.rs [hunk 2]
```diff
@@ -39,12 +39,22 @@ where
             if result.is_empty() {
                 debug!("No response");
             }
+            let results: i64 = result.len().try_into().unwrap();
             for response in result.into_iter() {
                 debug!(?response);
+                let controls =
```
Comment (nitnelave) [inline server/src/infra/ldap_server.rs:45]: If I wanted to be pedantic, I would ask you to assert that it's the last element in the response.

Thankfully, I'll have to maintain that code, and the simpler the better ;) I like this solution.

**Review by nitnelave — APPROVED** [review]
LGTM!

Can you test it with your setup to make sure that it's accepted?

**Comment by nitnelave** [comment]
You'll need to run `cargo fmt`: https://github.com/lldap/lldap/actions/runs/6420405003/job/17432425224?pr=699

File: server/src/infra/ldap_handler.rs (hunk 1 above)
Comment (broeng) [reply server/src/infra/ldap_handler.rs:192]: Good point, was thinking it'd be necessary, but can't really imagine it is. :)

**Comment by broeng** [comment]
I have pushed a formatted update, that also removes it from the listed features.

I'll get it tested as soon as possible (hopefully tomorrow) -- let's wait with merge until I can verify it solves the problem. :-)

**Comment by broeng** [comment]
> Can you test it with your setup to make sure that it's accepted?

Tested, and solves the/my problem. :+1: :-)

**Review by nitnelave — APPROVED** [review]

**Comment by nitnelave** [comment]
Thanks! Merged. It'll be in the `latest` image in a few minutes

**Pull request merged**